pkg/maps/configmap/configmap.go

```diff
@@ -0,0 +1,195 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"bytes"
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/common/types"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/lxcmap"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-config")
+
+const (
+	// MapNamePrefix is the basename prefix of endpoint config maps.
+	MapNamePrefix = "cilium_ep_config_"
+
+	// MaxEntries represents the maximum number of elements in the map
+	MaxEntries = 1
+
+	// SkipPolicyIngress causes ingress policy to be skipped.
+	SkipPolicyIngress = 1 << 0
+	// SkipPolicyEgress causes ingress policy to be skipped.
+	SkipPolicyEgress = 1 << 1
+)
+
+var (
+	flagsToString = map[int]string{
+		SkipPolicyIngress: "SKIP_POLICY_INGRESS",
+		SkipPolicyEgress:  "SKIP_POLICY_EGRESS",
+	}
+
+	binMap = map[uint]string{
+		0: "POLICY_INGRESS",
+		1: "POLICY_EGRESS",
+	}
+)
+
+// Flags is a set of endpoint configuration flags interpreted by BPF code.
+//
+// Must be in sync with the enum ep_cfg_flag in <bpf/lib/eps.h>
+type Flags uint32
+
+// String converts the specified flags into a human-readable form.
+func (f Flags) String() string {
+	var buffer bytes.Buffer
+	for i := uint(0); i < 32; i++ {
+		bitVal := int(f & (1 << i))
+		if bitVal > 0 {
+			if flag, ok := flagsToString[bitVal]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+				continue
+			}
+		} else if bitVal == 0 {
+			if flag, ok := binMap[i]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+				continue
+			}
+		}
+		buffer.WriteString(fmt.Sprintf("%04x,", 1<<i))
+	}
+
+	return buffer.String()
+}
+
+// Key is the key used to index into the config map for an endpoint.
+//
+// Must be in sync with the key of CONFIG_MAP in <bpf/lib/maps.h>
+type Key struct {
+	Bits uint32
+}
+
+func (k Key) String() string {
+	return fmt.Sprintf("%d", k)
+}
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// NewValue returns a new empty instance of the structure representing the BPF
+// map value
+func (k *Key) NewValue() bpf.MapValue { return &EndpointConfig{} }
+
+func (cfg *EndpointConfig) String() string {
+	// TODO - use a tabwriter for CLI consistency?
+	return fmt.Sprintf("%s, %d, %d, %s, %s, %d, %d, %s", cfg.Flags.String(), cfg.SecurityIdentity, cfg.SecurityIdentityNB, cfg.IPv4.String(), cfg.IPv6.String(), cfg.LXCID, cfg.LXCIDNB, cfg.NodeMAC.String())
+}
+
+// EndpointConfig represents the value of the endpoint's BPF map.
+//
+// Must be in sync with struct config_value in <bpf/lib/common.h>
+type EndpointConfig struct {
```

Comment:
Do we need to add some code somewhere for verifying that this structure matches the C version?

Reply:
Did this in `pkg/alignchecker` in newest commit.